easybuild/tools/github.py

```diff
@@ -395,6 +407,235 @@ def post_comment_in_issue(issue, txt, repo=GITHUB_EASYCONFIGS_REPO, github_user=
+
+    # add remote to pull from
+    github_target_repo = build_option('github_target_repo')
+    github_url = 'git@github.com:%s/%s.git' % (target_account, github_target_repo)
+    _log.debug("Cloning from %s", github_url)
+
+    origin = git_repo.create_remote('origin', github_url)
+    if not origin.exists():
+        raise EasyBuildError("%s does not exist?", github_url)
+
+    # git fetch
+    if start_branch is None:
+        start_branch = build_option('github_target_branch')
+    _log.debug("Fetching branch %s for remote %s", start_branch, origin)
+    origin.fetch(start_branch, depth=1)
+
+    # git checkout -b <branch>; git pull
+    if hasattr(origin.refs, start_branch):
+        origin_branch = getattr(origin.refs, start_branch)
+    else:
+        raise EasyBuildError("Branch '%s' not found at %s", start_branch, github_url)
+
+    _log.debug("Checking out branch %s from remote %s", start_branch, github_url)
+    git_repo.create_head(start_branch, origin_branch).checkout()
+    origin.pull(start_branch)
+
+    # copy files to right place
+    file_info = _copy_easyconfigs_to_repo(paths, git_working_dir)
+
+    # create target branch, if it was specified
+    if pr_branch:
+        git_repo.create_head(pr_branch).checkout()
+        _log.info("New branch '%s' created to commit files to", pr_branch)
+    else:
+        pr_branch = start_branch
+
+    # stage & commit files
+    for path, new in zip(file_info['paths_in_repo'], file_info['new']):
+        _log.debug("Staging and committing easyconfig %s (new: %s)", path, new)
+        git_repo.index.add([path])
+        # if no commit message is specified, commit changed files one-by-one
+        if commit_msg is None:
+            if new:
+                action = 'add'
+            else:
+                action = 'modify'
+            git_repo.index.commit("%s easyconfig %s" % (action, os.path.basename(path)))
+
+    # commit all staged in one go if a commit message is specified
+    if commit_msg is not None:
+        git_repo.index.commit(commit_msg)
+
+    # push to GitHub
+    github_user = build_option('github_user')
+    github_url = 'git@github.com:%s/%s.git' % (github_user, github_target_repo)
+    my_remote = git_repo.create_remote(github_user, github_url)
+    _log.info("Pushing branch %s to remote %s", pr_branch, my_remote)
+    my_remote.push(pr_branch)
+
+    return file_info, git_repo
+
+
+@only_if_module_is_available('git', pkgname='GitPython')
+def new_pr(paths, title=None, descr=None, commit_msg=None):
+    """Open new pull request using specified files."""
+    _log.info("Opening new pull request for  with %s", paths)
+
+    github_target_account = build_option('github_target_account')
+    github_target_repo = build_option('github_target_repo')
+    if github_target_repo != GITHUB_EASYCONFIGS_REPO:
+        raise EasyBuildError("Don't know how to open a pull request to the %s repository yet", github_target_repo)
```

Comment:
the main problem is that we don't know how to move the files supplied to the right place;

also, the way we determine the default branch name assumes we're dealing with easyconfigs, etc.

so, maybe this check should be moved to `_easyconfigs_pr_common`, I'll do that